components/consensusmanager/src/batch.rs

```diff
@@ -0,0 +1,30 @@
+use kaspa_consensus_core::{api::BlockValidationFuture, block::Block};
+use std::fmt::Debug;
+
+pub struct BlockProcessingBatch {
```

Comment:
I think it's more readable if you wrap the whole thing with an `Option`, instead of internal fields?

Reply:
Sometimes only one of the task vecs is taken and the others remain for future processing